irmago_test.go

```diff
@@ -985,6 +985,371 @@ func TestWizardFromScheme(t *testing.T) {
 	)
 }
 
+func TestWizardValidation(t *testing.T) {
+	conf := &Configuration{
+		CredentialTypes: map[CredentialTypeIdentifier]*CredentialType{
+			credid("scheme.issuer.a"): credtype("scheme.issuer.a"),
+			credid("scheme.issuer.b"): credtype("scheme.issuer.b",
+				"scheme.issuer.a",
+			),
+			credid("scheme.issuer.c"): credtype("scheme.issuer.c",
+				"scheme.issuer.a", "scheme.issuer.b",
+			),
+			credid("scheme.issuer.d"): credtype("scheme.issuer.d",
+				"scheme.issuer.a",
+			),
+			credid("scheme.issuer.e"): credtype("scheme.issuer.e",
+				"scheme.issuer.d",
+			),
+		},
+	}
+
+	wizard := IssueWizard{
+		ID: "testwizard",
+		Contents: IssueWizardContents{
+			{{
+				credwizarditem("scheme.issuer.c"),
+			}},
+			{{{
+				Type:       IssueWizardItemTypeCredential,
+				Credential: credidptr("scheme.issuer.e"),
+				Text:       &TranslatedString{"en": "custom description of credential e"},
+			}}},
+			{{{
+				Type:   IssueWizardItemTypeWebsite,
+				Header: &TranslatedString{"en": "header"},
+				Label:  &TranslatedString{"en": "label"},
+				Text:   &TranslatedString{"en": "text"},
+				URL:    &TranslatedString{"en": "https://example.com"},
+			}}},
+		},
+	}
+
+	err := wizard.Validate(conf)
+	require.NoError(t, err)
+}
+
+func TestWizardIncorrectContentsOrder(t *testing.T) {
+	conf := &Configuration{
+		CredentialTypes: map[CredentialTypeIdentifier]*CredentialType{
+			credid("scheme.issuer.a"): credtype("scheme.issuer.a"),
+			credid("scheme.issuer.b"): credtype("scheme.issuer.b",
+				"scheme.issuer.a",
+			),
+			credid("scheme.issuer.c"): credtype("scheme.issuer.c",
+				"scheme.issuer.a", "scheme.issuer.b",
+			),
+			credid("scheme.issuer.d"): credtype("scheme.issuer.d",
+				"scheme.issuer.a",
+			),
+			credid("scheme.issuer.e"): credtype("scheme.issuer.e",
+				"scheme.issuer.d",
+			),
+		},
+	}
+
+	wizard := IssueWizard{
+		ID: "testwizard",
+		Contents: IssueWizardContents{
+			{{
+				credwizarditem("scheme.issuer.c"),
+			}},
+			{{{
+				Type:   IssueWizardItemTypeWebsite,
+				Header: &TranslatedString{"en": "header"},
+				Label:  &TranslatedString{"en": "label"},
+				Text:   &TranslatedString{"en": "text"},
+				URL:    &TranslatedString{"en": "https://example.com"},
+			}}},
+			{{{
+				Type:       IssueWizardItemTypeCredential,
+				Credential: credidptr("scheme.issuer.e"),
+				Text:       &TranslatedString{"en": "custom description of credential e"},
+			}}},
+		},
+	}
+
+	err := wizard.Validate(conf)
+	require.Equal(t, "non-credential types in wizard testwizard should come last", err.Error())
```

Comment:
For requiring errors with specific error messages you can use `require.EqualError()` (below too).